pkg/audit/containment_test.go

```diff
@@ -106,31 +115,62 @@ func TestContainDelete(t *testing.T) {
 			ReverifyCount:     0,
 		}
 
-		err := planet.Satellites[0].DB.Containment().IncrementPending(ctx, info1)
+		err := containment.IncrementPending(ctx, info1)
 		require.NoError(t, err)
 
-		// check contained flag set to true
-		node, err := planet.Satellites[0].DB.OverlayCache().Get(ctx, info1.NodeID)
-		require.NoError(t, err)
-		require.True(t, node.Contained)
-
-		isDeleted, err := planet.Satellites[0].DB.Containment().Delete(ctx, info1.NodeID)
+		// delete the node from containment db
+		isDeleted, err := containment.Delete(ctx, info1.NodeID)
 		require.NoError(t, err)
 		require.True(t, isDeleted)
 
 		// check contained flag set to false
-		node, err = planet.Satellites[0].DB.OverlayCache().Get(ctx, info1.NodeID)
+		node, err := cache.Get(ctx, info1.NodeID)
 		require.NoError(t, err)
 		require.False(t, node.Contained)
 
 		// get pending audit that doesn't exist
-		_, err = planet.Satellites[0].DB.Containment().Get(ctx, info1.NodeID)
+		_, err = containment.Get(ctx, info1.NodeID)
 		require.Error(t, err, audit.ErrContainedNotFound.New(info1.NodeID.String()))
 		require.True(t, audit.ErrContainedNotFound.Has(err))
 
 		// delete pending audit that doesn't exist
-		isDeleted, err = planet.Satellites[0].DB.Containment().Delete(ctx, info1.NodeID)
+		isDeleted, err = containment.Delete(ctx, info1.NodeID)
 		require.NoError(t, err)
 		require.False(t, isDeleted)
 	})
 }
+
+func TestContainUpdateStats(t *testing.T) {
+	testplanet.Run(t, testplanet.Config{
+		SatelliteCount: 1, StorageNodeCount: 1,
+	}, func(t *testing.T, ctx *testcontext.Context, planet *testplanet.Planet) {
+		containment := planet.Satellites[0].DB.Containment()
+		cache := planet.Satellites[0].DB.OverlayCache()
+
+		info1 := &audit.PendingAudit{
+			NodeID:            planet.StorageNodes[0].ID(),
+			PieceID:           storj.PieceID{},
+			StripeIndex:       0,
+			ShareSize:         0,
+			ExpectedShareHash: pkcrypto.SHA256Hash(testrand.Bytes(10)),
+			ReverifyCount:     0,
+		}
+
+		err := containment.IncrementPending(ctx, info1)
+		require.NoError(t, err)
+
+		// update node stats
+		_, err = cache.UpdateStats(ctx, &overlay.UpdateRequest{NodeID: info1.NodeID})
+		require.NoError(t, err)
+
+		// check contained flag set to false
+		node, err := cache.Get(ctx, info1.NodeID)
+		require.NoError(t, err)
+		require.False(t, node.Contained)
```

Comment:
We could use an assert because it doesn't stop the test to continue with the next check and at the end of the test we would know in one run if this fails but also the next one `require.True(t, audit.ErrContainedNotFound.Has(err))` too

Reply:
I changed some requires to be asserts in the latest commit.